lib/modules/manager/github-actions/extract.ts

```diff
@@ -226,6 +229,44 @@ function extractWithYAMLParser(
     }
 
     deps.push(...extractRunners(job?.['runs-on']));
+
+    const actionsWithVersions: Record<string, Partial<PackageDependency>> = {
+      go: {
+        versioning: npmVersioning.id,
+      },
+      node: {
+        versioning: nodeVersioning.id,
+      },
+      python: {
+        versioning: npmVersioning.id,
```

Comment:
this will cause issues for some inputs 🤔 

https://github.com/actions/setup-python?tab=readme-ov-file#basic-usage

Reply:
I know. It seems GitHub are using npm semver syntax for all of these. We maybe need to check their source

Reply:
Added readme to clarify. I think there's little harm getting this out and then getting feedback for any "invalid value" we find. I don't think it will do any _wrong_ updates